### PR TITLE
Discoverd deployment fixes

### PR DESCRIPTION
**Pull request by josephglanville**
These are some preliminary fixes to issues that can occur during discoverd deployments.
The main one is the addition of the deployment ID to the discoverd metadata as it fixes stale reads of service meta.

### CODE REVIEW

File: discoverd/main.go [hunk 1]
```diff
@@ -223,8 +225,18 @@ func (m *Main) Run(args ...string) error {
...
 	}
 	m.Notify(opt.Notify, "http://"+httpAddr, opt.DNSAddr)
-	go discoverd.NewClientWithURL("http://"+httpAddr).AddServiceAndRegister("discoverd", httpAddr)
-
+	go func() {
+		for {
+			hb, err := discoverd.NewClientWithURL("http://"+httpAddr).AddServiceAndRegister("discoverd", httpAddr)
+			if err != nil {
+				continue
+			}
```
Comment (titanous) [inline discoverd/main.go]: This should probably log/sleep?

Comment (josephglanville) [inline discoverd/main.go]: Indeed, fixed.

File: discoverd/main.go [hunk 2]
```diff
@@ -223,8 +225,20 @@ func (m *Main) Run(args ...string) error {
 		httpAddr = net.JoinHostPort(os.Getenv("EXTERNAL_IP"), port)
 	}
 	m.Notify(opt.Notify, "http://"+httpAddr, opt.DNSAddr)
-	go discoverd.NewClientWithURL("http://"+httpAddr).AddServiceAndRegister("discoverd", httpAddr)
-
+	go func() {
+		for {
+			hb, err := discoverd.NewClientWithURL("http://"+httpAddr).AddServiceAndRegister("discoverd", httpAddr)
+			if err != nil {
+				m.logger.Println("failed to register service/instance, retrying in 5 seconds")
```
Comment (titanous) [inline discoverd/main.go]: This should log the error.

**Comment by titanous** [comment]
LGTM

**Comment by titanous** [comment]
LGTM

**Pull request merged**